src/main/kotlin/org/web3j/evm/EmbeddedEthereum.kt

```diff
@@ -95,16 +103,19 @@ class EmbeddedEthereum(configuration: Configuration, private val operationTracer
         protocolSchedule = MainnetProtocolSchedule.fromConfig(genesisConfig.configOptions)
         genesisState = GenesisState.fromConfig(genesisConfig, protocolSchedule)
 
-        val storageProvider = KeyValueStorageProvider(
-            InMemoryKeyValueStorage(),
-            InMemoryKeyValueStorage(),
-            InMemoryKeyValueStorage(),
+        val storageProvider = KeyValueStorageProvider({ InMemoryKeyValueStorage() },
             InMemoryKeyValueStorage(),
             false
         )
         val metricsSystem = NoOpMetricsSystem()
 
-        protocolContext = ProtocolContext.init(storageProvider, genesisState, protocolSchedule, metricsSystem) { _, _ -> null }
+        val hashValueStore: Map<Bytes, ByteArray> = HashMap()
+        val stateStorage: InMemoryKeyValueStorage = TestInMemoryStorage(hashValueStore)
+        val worldStateStorage: WorldStateStorage = WorldStateKeyValueStorage(stateStorage)
+        val worldStateArchive: WorldStateArchive = DefaultWorldStateArchive(
+            worldStateStorage, WorldStatePreimageKeyValueStorage(InMemoryKeyValueStorage())
+        )
+        protocolContext = ProtocolContext.init(storageProvider, worldStateArchive, genesisState, protocolSchedule, metricsSystem, { storageProvider, worldStateArchive -> }, 6L)
```

Comment:
We are not using these : `storageProvider, worldStateArchive`, we can change them by `_`